tests/integration_tests/datasets/commands_tests.py

```diff
@@ -586,6 +586,21 @@ def test_create_dataset_command(self, mock_g, mock_g2):
             engine.execute("DROP TABLE test_create_dataset_command")
         db.session.commit()
 
+    @patch("superset.security.manager.g")
+    @patch("superset.commands.utils.g")
+    def test_create_dataset_command_not_allowed(self, mock_g, mock_g2):
+        mock_g.user = security_manager.find_user("gamma")
+        mock_g2.user = mock_g.user
```

Comment:
Instead of mocking, couldn't we use `with override_user(user)` here?

Reply:
changed, and changed all on security_tests.py also